openpype/client/server/conversion_utils.py

```diff
@@ -584,7 +584,9 @@ def representation_fields_v3_to_v4(fields, con):
     representation_attributes = con.get_attributes_for_type("representation")
 
     output = set()
-    for field in fields:
+    fields = set(fields)
+    iterfields = fields.copy()
+    for field in iterfields:
```

Comment:
Why is this needed?

Reply:
I believe it was to resolve this bug reported here: https://github.com/ynput/OpenPype/issues/4949#issue-1701076658 which fails on [this line](https://github.com/ynput/OpenPype/blob/release/next-minor/openpype/client/server/conversion_utils.py#L603-L606) due to `fields` not being a `set`. 

A copy is likely made since it appears that the `fields` variable is being update as it is being iterated over. So it should basically be this:
```python
fields = set(fields)
for field in fields.copy():
    ...
```

Reply:
Isn't the bug that there should be `output |= ...` instead of copying source fields?